catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/fragment/ScriptFragment.kt

```diff
@@ -0,0 +1,1009 @@
+        listView = view.findViewById(android.R.id.list)
+        scriptFinder = view.findViewById(R.id.findview)
+
+        scriptFinder?.setOnResultFoundListener(object : OnResultFoundListener {
+            override fun onResultFound(
+                sceneIndex: Int,
+                spriteIndex: Int,
+                brickIndex: Int,
+                totalResults: Int,
+                textView: TextView?
+            ) {
+                val currentProject =
+                    projectManager.currentProject
+                val currentScene =
+                    currentProject.sceneList[sceneIndex]
+                val currentSprite =
+                    currentScene.spriteList[spriteIndex]
+                textView?.text = createActionBarTitle(
+                    currentProject,
+                    currentScene,
+                    currentSprite
+                )
+                projectManager.setCurrentSceneAndSprite(
+                    currentScene.name,
+                    currentSprite.name
+                )
+                adapter.updateItems(currentSprite)
+                adapter.notifyDataSetChanged()
+                listView?.smoothScrollToPosition(brickIndex)
+                highlightBrickAtIndex(brickIndex)
+                hideKeyboard()
+            }
+        })
+
+        val activity = activity as SpriteActivity?
+
+        scriptFinder?.setOnCloseListener(object : OnCloseListener {
+            override fun onClose() {
+                listView?.cancelHighlighting()
+                finishActionMode()
+                if (activity != null && !activity.isFinishing) {
+                    activity.setCurrentSceneAndSprite(
+                        projectManager.currentlyEditedScene,
+                        projectManager.currentSprite
+                    )
+                    activity.supportActionBar?.title = activity.createActionBarTitle()
+                    activity.addTabs()
+                }
+                activity?.findViewById<View>(R.id.toolbar)?.visibility = View.VISIBLE
+            }
+        })
+
+        scriptFinder?.setOnOpenListener(object : OnOpenListener {
+            override fun onOpen() {
+                activity?.removeTabs()
+                activity?.findViewById<View>(R.id.toolbar)?.visibility = View.GONE
+            }
+        })
+
+        setHasOptionsMenu(true)
+        SnackbarUtil.showHintSnackbar(getActivity(), R.string.hint_scripts)
+        return view
+    }
+
+    fun createActionBarTitle(
+        currentProject: Project,
+        currentScene: Scene,
+        currentSprite: Sprite
+    ): String {
+        return if (currentProject.sceneList.size == 1) {
+            currentSprite.name
+        } else {
+            currentScene.name + ": " + currentSprite.name
+        }
+    }
+
+    private fun highlightBrickAtIndex(index: Int) {
+        listView?.getBrickPositionsToHighlight()?.clear()
```

Comment:
please use property access 
```suggestion
        listView?.brickPositionsToHighlight?.clear()
```